packages/x-charts/src/internals/filterAttributeSafeProperties.ts

```diff
@@ -0,0 +1,37 @@
+const propNames = [
```

Comment:
```suggestion
const propNames = new Set([
```

You are looping through the array a lot, a `Set` would be a lot more performant 👍